lib/fs.js

```diff
@@ -1615,6 +1615,11 @@ function ReadStream(path, options) {
   if (!(this instanceof ReadStream))
     return new ReadStream(path, options);
 
+  if (typeof options === 'string')
+    options = { encoding: options };
+  else if (options && typeof options !== 'object')
+    throw new TypeError('Bad arguments');
```

Comment:
Something like `options must be a string or an object` is simple enough I think. Other than that, LGTM too.